dask/base.py

```diff
@@ -839,7 +839,17 @@ def normalize_object(o):
     method = getattr(o, "__dask_tokenize__", None)
     if method is not None:
         return method()
-    return normalize_function(o) if callable(o) else uuid.uuid4().hex
+
+    if callable(o):
+        return normalize_function(o)
+
+    if config.get("tokenize.allow-random", True):
+        return uuid.uuid4().hex
+
+    raise RuntimeError(
+        f"Object {str(o)} doesn't have a `__dask_tokenize__` method and "
+        "cannot be normalized deterministically."
+    )
```

Comment:
Just a small suggestion on the error message here. In particular, it would be nice to point users to a relevant section in the docs. 

```
Object {str(o)} cannot be deterministically hashed. Please see
https://docs.dask.org/en/latest/custom-collections.html#implementing-deterministic-hashing
for more information
```

Reply:
Yep, I'll update it when I get a chance.